earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp

```diff
@@ -23,14 +23,343 @@ namespace {
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dashInValue) {
+    const std::string js = R"js_({name:"fred-flintstone"})js_";
+    const std::string expectedJson = R"js_({"name":"fred-flintstone"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, spaceStartValue) {
+    const std::string js = R"js_({" name":"fred"})js_";
+    const std::string expectedJson = R"js_({" name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// This is legal in json
+TEST_F(JsonUnitTest, underscoreStartValue) {
+    const std::string js = R"js_({"_name":"fred"})js_";
+    const std::string expectedJson = R"js_({"_name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dotStartValue) {
+    const std::string js = R"js_({".name":"fred"})js_";
+    const std::string expectedJson = R"js_({".name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dashStartValue) {
+    const std::string js = R"js_({"-name":"fred"})js_";
+    const std::string expectedJson = R"js_({"-name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// While illegal in json with . reference it is ok otherwise
+TEST_F(JsonUnitTest, numericStartValue) {
+    const std::string js = R"js_({"123name":"fred"})js_";
+    const std::string expectedJson = R"js_({"123name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Simple multiple pair case
+TEST_F(JsonUnitTest, multiplePairs) {
+    const std::string js = R"js_({name:"fred",car:"christine"})js_";
+    const std::string expectedJson = 
+        R"js_({"name":"fred","car":"christine"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Should it eat white space? I would think the conversion would collapse it.
+// This may not be a bug.  Yes the test is ugly on purpose to contrast output
+//TEST_F(JsonUnitTest, spansMultipleLines) {
+//    const std::string js = R"js_({name:"fred", 
+//         car:"christine"
+//        ,color:"red"})js_";
+//    const std::string expectedJson = 
+//        R"js_({"name":"fred","car":"christine","color":"red"})js_";
+//    std::string result = JsonUtils::JsToJson(js);
+//
+//    EXPECT_EQ(expectedJson, result);
+//}
+
+// Return/NewLine should be... eaten? Replaced with a space?
+// Surely not just have extra escape chars thrown at it
+//TEST_F(JsonUnitTest, newlineInValue) {
+//    const std::string js = R"js_({"name":"fred\r\nis\r\ncool\r\n"})js_";

```

Comment:
In this case you are stating you want new lines in your data.  This is perfectly allowed and should be transformed to JSON...  I'm wondering with the raw string you could just add literal new lines.  But in any case it should be preserved in the JSON too.

Curious what this would do:

```
R"js_({"name":"fred
is
cool
"})js_"
```

It should produce the same string for JSON (with the literal new lines).  I'm not sure how to do return carriages in a raw string as all characters are taken literally.